delft3dgtprotectedservices/views.py

```diff
@@ -24,8 +25,13 @@ def files(request, simulation_uuid, loc):
     # get scene or 404
     scene = get_object_or_404(Scene, suid=uuid)
 
+    loc_path, loc_extension = os.path.splitext(loc)
+
     # return 403 if not allowed
-    if not request.user.has_perm("view_scene", scene):
+    if not request.user.has_perm("restricted_view_scene", scene):
+        return HttpResponse(status=403)
+
+    if not request.user.has_perm("view_scene", scene) and loc_extension != '.png':
```

Comment:
These are switched now right? If you have the `view_scene` permission, you can view everything. The `restricted_view_scene` is only allowed to view pictures.

If you switch them, the negatives won't work. Maybe better, also to clear up to what the `not` applies to, to write it in a positive manner (if user as permission x or permission y do x else forbid).

Reply:
I think the logic is correct:
- if you don't have restricted_view permission -> 403
- in the second if (we know we have restricted_view permissions) is you don't have view permissions and extenstions is not png -> 403
- else return 200 with response.

Tests are succeeding, so if the logic is not correct, what is then problem in the tests?